lib/Rx/Observable/ForkJoinObservable.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Rx\Observable;
+
+use Rx\Observable;
+use Rx\Observer\AutoDetachObserver;
+use Rx\Observer\CallbackObserver;
+use Rx\ObserverInterface;
+use Rx\Disposable\CompositeDisposable;
+use Rx\Scheduler\ImmediateScheduler;
+use Rx\SchedulerInterface;
+
+class ForkJoinObservable extends Observable {
+
+    /**
+     * @var Observable[]
+     */
+    private $observables;
+
+    private $values = [];
+
+    private $completed = 0;
+
+    private $resultSelector;
+
+    public function __construct(array $observables = [], callable $resultSelector = null) {
```

Comment:
$observables is optional here, but required on Observable::forkJoin.  It should probably be required everywhere.